opentracing-api/src/main/java/io/opentracing/propagation/Binary.java

```diff
@@ -33,10 +33,11 @@
  */
 public interface Binary {
     /**
-     * Writes a sequence of bytes to this channel from the given buffer.
+     * Writes a sequence of bytes to this carrier from the given buffer.
      * The internal buffer is expected to grow as more data is written.
      *
-     * The behavior of this method is expected to be the same as WritableByteChannel.write().
+     * The behavior of this method is expected to be the same as WritableByteChannel.write(),
+     * with the exception of being synchronous.
```

Comment:
Not sure if this is accurate, I think we want to make it return void. I.e. either write all content or throw exception.

Reply:
A single operation then? That would work just fine actually. That would simplify things a lot. The question would be what to do for `read()` then - throw an `IOException` if the passed `ByteBuffer` is too small (as you mentioned in the mentioned discussion)?

(Returning a `ByteBuffer` from `read()` wouldn't be so nice IHMO ;) )

Reply:
no, it can still be multiple calls if the tracer's buffer is smaller than the serialized context. The point is that the carrier must accept all or nothing. But if you want to keep returning int it's fine too.

I'd rather punt on this whole thing in 0.31. Yes, its going to be a breaking change in the next release, but nobody uses the existing binary API anyway.

Reply:
After prototyping stuff a little bit, I agree the best of is to expect the `ByteBuffer` to be written *all* (this `void write(...)`), even if we allow multiple calls to it. Will update this PR earlier today with an update to `Adapters` ;)